dvc/remote/local/__init__.py

```diff
@@ -128,7 +128,14 @@ def link(self, cache_info, path_info):
         cache = cache_info["path"]
         path = path_info["path"]
 
-        assert os.path.isfile(cache)
+        if not os.path.isfile(cache):
```

Comment:
You've removed a generic logic from base and only left part of it in RemoteLOCAL. Doesn't seem right :slightly_smiling_face: